all(ci): Fix failing integration tests

GodTamIt opened the pull request:
This adds the `--wait` and `--wait-timeout` options to the `docker compose up` so that containers are ready before running tests.

This also bumps `storj/up` to v1.2.10, which adds the fix to `versioncheck` (removes it), so that the integration test environment successfully loads up.

Fixes #96

ifraixedes committed:
uplink(fix): Resolve linter issues

Fix 2 linter issues that weren't caught by the previous Clippy version
that we used.

GodTamIt committed:
all(ci): Wait for Docker compose to be ready

This adds the `--wait` and `--wait-timeout` options to the
`docker compose up` so that containers are ready before running tests.

GodTamIt committed:
all(deps): Bump `storj/up` to v1.2.10

This includes version v1.2.8, which removes `versioncontrol`, which was
causing CI failures.

GodTamIt commented:
@ifraixedes This looks to fix the integration tests now.

The `check-implementation` was still failing because of #93, but I've pulled that into this change.

ifraixedes reviewed (commented):
@GodTamIt thank you very much

Could you update commit messages?
- [`dd73f4f` (#97)](https://github.com/storj-thirdparty/uplink-rust/pull/97/commits/dd73f4f634d8ece2fae75ba04cf13a46c6f46a0c) subject to be prefixed with `all(deps):`
- [`7aebe94` (#97)](https://github.com/storj-thirdparty/uplink-rust/pull/97/commits/7aebe949826538771d97ac212b938e6138f05c0a) to rename the previs to `all(ci):`

GodTamIt committed:
uplink(fix): Resolve new `io::Error` lints

With the new version of Rust Clippy, these lints now suggest using
`io::Error::other` for construction of `other` `io::Error` variants.
